src/cmap/connection_pool_events.ts

```diff
@@ -54,14 +54,19 @@ export abstract class ConnectionPoolMonitoringEvent {
  */
 export class ConnectionPoolCreatedEvent extends ConnectionPoolMonitoringEvent {
   /** The options used to create this connection pool */
-  options?: ConnectionPoolOptions;
+  options: Pick<
+    ConnectionPoolOptions,
+    'maxPoolSize' | 'minPoolSize' | 'maxConnecting' | 'maxIdleTimeMS' | 'waitQueueTimeoutMS'
+  >;
```

Comment:
Since we're not consuming this API we don't have a type check here if ConnectionPoolOptions is changed it will just automatically be pulled in, can we add a type test?

Reply:
Although we do it in other places, perhaps we shouldn't use `Pick` in our public APIs.  Generally our TS APIs should be as static as possible (i.e., hard coded) to ensure we don't break them by accident.  If the goal here was to ensure that `options`' properties types always match `ConnectionPoolOptions`, then could we consider something like this instead?  This would require no type tests.

```typescript
options: ConnectionPoolOptions & { maxPoolSize: number, minPoolSize: number, maxConnecting: number, ... }
```

Reply:
Isn't that sample code different? IIUC you are including the `ConnectionPoolOptions` again by intersecting the two types. Changing the declaration to that format does not remove the usefulness of type tests, they codify the public API as a downstream user would consume them, this format would conveniently warn on type changes but it is a distinct scenario we are trying to cover.